### PR TITLE
update dep

### PR DESCRIPTION
**Pull request by ebuchman**
We actually shouldn't merge this as is - I updated tmlibs and abci and not sure why the state tests are now failing. Seems to be because of a super minor mismatch in empty struct in the KI64Pair between `[]uint{}` and `[]uint(nil)`  :(

Maybe @xla or @melekes can help figure this out .

See the main change of interest: https://github.com/tendermint/abci/compare/v0.10.0...v0.10.2#diff-f325a92d451852ea8923aaa3a50731beR167

### CODE REVIEW

**Comment by melekes** [comment]
```
expected: &types.ResponseDeliverTx{Code:0x0, Data:[]uint8{0x62, 0x61, 0x72}, Log:"ok", Info:"", GasWanted:0, GasUsed:0, Tags:[]common.KVPair(nil), Fee:common.KI64Pair{Key:[]uint8(nil), Value:0}}
                        actual  : &types.ResponseDeliverTx{Code:0x0, Data:[]uint8{0x62, 0x61, 0x72}, Log:"ok", Info:"", GasWanted:0, GasUsed:0, Tags:[]common.KVPair{}, Fee:common.KI64Pair{Key:[]uint8{}, Value:0}}
```

**Comment by melekes** [comment]
```
--- FAIL: TestPEXReactorRunning (0.32s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x74ccf4]

goroutine 69 [running]:
testing.tRunner.func1(0xc4202121e0)
	/usr/local/go/src/testing/testing.go:742 +0x567
panic(0xa3c380, 0xeca2b0)
	/usr/local/go/src/runtime/panic.go:505 +0x24a
github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common.(*RepeatTimer).Stop(0x0)
	/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common/repeat_timer.go:177 +0x34
github.com/tendermint/tendermint/p2p/conn.(*MConnection).OnStop(0xc420218240)
	/go/src/github.com/tendermint/tendermint/p2p/conn/connection.go:205 +0x85
github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common.(*BaseService).Stop(0xc420218240, 0x2, 0x2)
	/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common/service.go:154 +0x29c
github.com/tendermint/tendermint/p2p.(*peer).OnStop(0xc420218120)
	/go/src/github.com/tendermint/tendermint/p2p/peer.go:200 +0x53
github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common.(*BaseService).Stop(0xc420218120, 0x984bd3, 0xc420227c80)
	/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common/service.go:154 +0x29c
github.com/tendermint/tendermint/p2p.(*Switch).OnStop(0xc42013e420)
	/go/src/github.com/tendermint/tendermint/p2p/switch.go:193 +0x16c
github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common.(*BaseService).Stop(0xc42013e420, 0x0, 0x0)
	/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/tmlibs/common/service.go:154 +0x29c
github.com/tendermint/tendermint/p2p/pex.TestPEXReactorRunning(0xc4202121e0)
	/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor_test.go:102 +0x893
testing.tRunner(0xc4202121e0, 0xbf8b78)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565
FAIL	github.com/tendermint/tendermint/p2p/pex	0.839s
```

**Comment by melekes** [comment]
Have no idea how this happened. Looks like `pingTimer` is `nil` by the time we're stopping it.

**Comment by melekes** [comment]
I think the bug is still there despite the tests passing.

**Comment by ebuchman** [comment]
Thanks @melekes  - let's open new issue/s for this ?

File: p2p/switch.go [hunk 1]
```diff
@@ -561,14 +563,17 @@ func (sw *Switch) addPeer(pc peerConn) error {
 	return nil
 }
 
-func (sw *Switch) startInitPeer(peer *peer) {
+func (sw *Switch) startInitPeer(peer *peer) error {
 	err := peer.Start() // spawn send/recv routines
 	if err != nil {
 		// Should never happen
 		sw.Logger.Error("Error starting peer", "peer", peer, "err", err)
+		return err
```
Comment (odeke-em) [inline p2p/switch.go]: Oops, this def looks like an oversight, we should have had this return statement even without the error.

**Pull request merged**